src/Money.php

```diff
@@ -769,6 +789,7 @@ public function convertedTo(
      */
     public function formatWith(\NumberFormatter $formatter) : string
     {
+        /** @psalm-suppress ImpureMethodCall */
```

Comment:
Does it make sense to suppress this?

As I understand it, this method is not pure as you could pass the same `NumberFormatter` instance twice but configured differently, and the method would return different results?

Reply:
this is bad news, because `formatWith()` will prevent `Money` from being a `@psalm-immutable` object otherwise :(